maitake/src/wait/cell.rs

```diff
@@ -102,24 +138,28 @@ impl WaitCell {
                     "state changed unexpectedly while parking!"
                 );
 
-                if let Some(prev_waker) = prev_waker {
-                    prev_waker.wake();
-                }
-
                 if let Some(waker) = waker {
                     waker.wake();
                 }
 
-                if test_dbg!(state.is(State::CLOSED)) {
-                    wait::closed()
-                } else {
-                    wait::notified(())
-                }
+                // We just went to the closed state, so sorry new waker, shops
+                // closed
+                closed()
             }
         }
     }
 
-    pub fn notify(&self) -> bool {
+    /// Wait to be woken up by this cell.
+    ///
+    /// Note: The waiter is not registered until AFTER the first time the waiter is polled
```

Comment:
i wonder if it would make sense to add something like `WaitMap::enqueue` here, for eagerly registering a `Wait` future? but we can add that in a follow-up PR.

Reply:
Agree, but will do in a follow up PR, kind of spent on this one ATM, and would like to get it merged.